__tests__/AppCatalog.js

```diff
@@ -22,162 +26,633 @@ import {
-    `/v4/clusters/${V4_CLUSTER.id}/`,
-    v4AWSClusterResponse
-  );
-  requests.status = getPersistedMockCall(
-    `/v4/clusters/${V4_CLUSTER.id}/status/`,
-    v4AWSClusterStatusResponse
-  );
-  requests.apps = getPersistedMockCall(
-    `/v4/clusters/${V4_CLUSTER.id}/apps/`,
-    appsResponse
-  );
-  requests.credentials = getPersistedMockCall(
-    `/v4/organizations/${ORGANIZATION}/credentials/`
-  );
-  requests.releases = getPersistedMockCall('/v4/releases/', releasesResponse);
-
-  requests.appcatalogs = getPersistedMockCall(
-    '/v4/appcatalogs/',
-    appCatalogsResponse
-  );
-
-  nock('https://cors-anywhere.herokuapp.com')
-    .get(/.*$/)
-    .reply(
-      StatusCodes.InternalServerError,
-      'Not supposed to get here in these tests.'
-    )
-    .persist();
-});
+describe('AppCatalog', () => {
+  // eslint-disable-next-line no-console
+  const originalConsoleError = console.error;
 
-// Stop persisting responses
-afterAll(() => {
-  Object.keys(requests).forEach(req => {
-    requests[req].persist(false);
+  beforeAll(() => {
+    nock.disableNetConnect();
+    // eslint-disable-next-line no-console
+    console.error = jest.fn();
   });
-});
 
-/************ TESTS ************/
-
-it('renders all non internal app catalogs in the app catalogs overview', async () => {
-  nock('https://catalogshost')
-    .get('/giantswarm-incubator-catalog/index.yaml')
-    .reply(StatusCodes.Ok, catalogIndexResponse);
-  nock('https://catalogshost')
-    .get('/giantswarm-test-catalog/index.yaml')
-    .reply(StatusCodes.Ok, catalogIndexResponse);
-  nock('https://catalogshost')
-    .get('/helmstable/index.yaml')
-    .reply(StatusCodes.Ok, catalogIndexResponse);
-
-  const { findByText } = renderRouteWithStore(AppCatalogRoutes.Home);
-
-  const introText = await findByText(
-    'Pick an App Catalog to browse all the Apps in it.'
-  );
-  expect(introText).toBeInTheDocument();
-
-  for (const catalog of appCatalogsResponse) {
-    // Skip expectation for internal catalogs.
-    // They should not show up in Happa.
-    if (
-      catalog.metadata.labels['application.giantswarm.io/catalog-type'] ===
-      'internal'
-    ) {
-      continue;
+  afterAll(() => {
+    nock.enableNetConnect();
+    // eslint-disable-next-line no-console
+    console.error = originalConsoleError;
+  });
+
+  beforeEach(() => {
+    getMockCall('/v4/user/', userResponse);
+    getMockCall('/v4/info/', AWSInfoResponse);
+    getMockCall('/v4/organizations/', orgsResponse);
+    getMockCall(`/v4/organizations/${ORGANIZATION}/`, orgResponse);
+    getMockCall('/v4/clusters/', v4ClustersResponse);
+    getMockCall(`/v4/clusters/${V4_CLUSTER.id}/`, v4AWSClusterResponse);
+    getMockCall(
+      `/v4/clusters/${V4_CLUSTER.id}/status/`,
+      v4AWSClusterStatusResponse
+    );
+    getMockCall(`/v4/organizations/${ORGANIZATION}/credentials/`);
+    getMockCall('/v4/appcatalogs/', appCatalogsResponse);
+  });
+
+  afterEach(() => {
+    expect(nock.isDone());
```

Comment:
I noticed we have this in a few places, we're not actually checking anything here 😅 

```suggestion
    expect(nock.isDone()).toBeTruthy()`;
```

or `toBe(true)` if you prefer.

getting strict on the requests doesn't have to be in the scope of this PR though if changing this causes you a lot more work right now.